services/src/layers/listing.rs

```diff
@@ -51,4 +136,40 @@ pub trait DatasetLayerCollectionProvider {
 
     /// get the full contents of the layer with the given `id`
     async fn load_dataset_layer(&self, id: &LayerId) -> Result<Layer>;
+
+    /// Get a list of supported search capabilities
+    async fn get_search_capabilities(&self) -> Result<SearchCapabilities> {
+        Ok(SearchCapabilities {
+            search_types: SearchTypes {
+                fulltext: false,
+                prefix: false,
+            },
+            autocomplete: false,
+            filters: None,
+        })
+    }
+
+    /// Perform a search
+    #[allow(unused_variables)]
+    async fn search(
```

Comment:
should this be in a separate trait, if it is optional to implement it?

Reply:
We need to get at least an error from each provider.
How should the method know if some of the providers implement a trait? Dyn-Downcasting by using the capabilities as reference?